numba_cuda/numba/cuda/tests/nrt/test_nrt.py

```diff
@@ -0,0 +1,108 @@
+import gc
+import numpy as np
+import unittest
+from unittest.mock import patch
+from numba.core.runtime import rtsys
+from numba.tests.support import TestCase, EnableNRTStatsMixin
+from numba.cuda.testing import CUDATestCase
+
+from numba import cuda
+
+from numba.core import errors, types
+from numba.core.extending import overload
+from numba.np.arrayobj import (_check_const_str_dtype, is_nonelike,
+                               ty_parse_dtype, ty_parse_shape, numpy_empty_nd)
+
+
```

Comment:
I think some explanation here would be good. When I came to this PR, I forgot what the point of this was and had to spend some time figuring it out. I think I wrote this workaround, and if I get confused by it, I think it definitely needs some explanation for others! E.g.:
```suggestion
# Typical tests for allocation use array construction (e.g. np.zeros, np.empty,
# etc.) to induce allocations. These don't work in the CUDA target because they
# need keyword arguments, which are presently not supported properly in the
# CUDA target.
#
# To work around this, we define the cuda_empty() function, that works like
# np.empty(), except that it uses only positional arguments.
#
# Once the CUDA target supports keyword arguments, this workaround will no
# longer be necessary and the tests in this module should be switched to use
# the relevant NumPy functions instead.

```

Reply:
What do you think of the changes in https://github.com/NVIDIA/numba-cuda/pull/17/commits/8e389f1eb7c376232bea25aca41d573e3e5e47e4 ?